frontend/public/components/modals/installplan-preview-modal.tsx

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable no-undef, no-unused-vars */
+
+import * as React from 'react';
+import { safeDump } from 'js-yaml';
+
+import { createModalLauncher, ModalTitle, ModalBody, ModalFooter } from '../factory/modal';
+import { StepResource, referenceForStepResource } from '../operator-lifecycle-manager';
+import { ResourceLink, CopyToClipboard } from '../utils';
+
+export const installPlanPreviewModal = createModalLauncher<InstallPlanPreviewModalProps>(({cancel, stepResource}) => <div className="modal-content">
+  <ModalTitle>Install Plan Preview <ResourceLink linkTo={false} name={stepResource.name} kind={referenceForStepResource(stepResource)} /></ModalTitle>
+  <ModalBody>
+    <CopyToClipboard value={safeDump(JSON.parse(stepResource.manifest))} />
```

Comment:
If this could ever be invalid JSON, we should handle `JSON.parse` errors

Reply:
It should never be invalid JSON because OLM uses this data to install the Operator. If it is invalid, we want to know about the error because that means OLM won't be able to resolve the resource.